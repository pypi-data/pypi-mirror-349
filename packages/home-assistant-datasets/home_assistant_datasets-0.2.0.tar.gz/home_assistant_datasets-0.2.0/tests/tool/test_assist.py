"""Test for running the assist eval end to end command."""

import pathlib
import tempfile

from syrupy import SnapshotAssertion

from . import run_cmd


def test_assist(snapshot: SnapshotAssertion) -> None:
    """Run the assist eval."""
    with tempfile.TemporaryDirectory() as tmpdir:
        run_cmd(
            [
                "pytest",
                "home_assistant_datasets/tool/assist/collect",
                "--models=assistant",
                "--dataset=./datasets/assist",
                f"--model_output_dir={tmpdir}",
                "--categories=light",
                "--count=1",
            ]
        )
        # Ensure the scrape succeeded before proceeding
        assert (pathlib.Path(tmpdir) / "assistant" / "_scrape_context.yaml").exists()

        run_cmd(
            [
                "pytest",
                "home_assistant_datasets/tool/assist/eval",
                f"--model_output_dir={tmpdir}",
            ]
        )
        output_files = sorted(list(pathlib.Path(tmpdir).glob("*.*")))
        assert [output.name for output in output_files] == snapshot
        for output in output_files:
            assert (output.name, output.read_text()) == snapshot
