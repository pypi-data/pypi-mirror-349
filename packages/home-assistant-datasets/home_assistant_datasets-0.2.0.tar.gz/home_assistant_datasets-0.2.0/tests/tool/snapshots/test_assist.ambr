# serializer version: 1
# name: test_assist
  list([
    'report.csv',
    'reports-by-category.yaml',
    'reports-by-model-category.yaml',
    'reports-by-model-test-name.yaml',
    'reports-by-task-id.yaml',
    'reports-by-test-name.yaml',
    'reports-token-stats.yaml',
    'reports.yaml',
  ])
# ---
# name: test_assist.1
  tuple(
    'report.csv',
    '''
      task_id,model_id,category,text,tool_call,response,task_name,label,details
      "dom1_pl_lights-kitchen_light_off-0","assistant","light","Kitchen light off","{'tool_name': None, 'tool_args': None}","Turned off the light","eval-test_expected_states","Good",""
      "dom1_pl_lights-please_turn_off_the_light-0","assistant","light","Please turn off the light","{'tool_name': None, 'tool_args': None}","Turned off the lights","eval-test_expected_states","Good",""
      "dom1_pl_lights-please_turn_on_the_kitchen_light-0","assistant","light","Please turn on the kitchen light","{'tool_name': None, 'tool_args': None}","Turned on the light","eval-test_expected_states","Good",""
      "dom1_pl_lights-shut_off_the_upstairs_lights_please-0","assistant","light","Shut off the upstairs lights please","","Sorry, I am not aware of any device called off the upstairs lights","eval-test_expected_states","Bad","AssertionError: assert not {'light.bedroom_2_light': {'expected': {'state': 'off'}, 'got': {'state': 'on'}}, 'light.bedroom_3_light': {'expected'...ate': 'off'}, 'got': {'state': 'on'}}, 'light.bedroom_4_light': {'expected': {'state': 'off'}, 'got': {'state': 'on'}}}"
      "dom1_pl_lights-turn_off_the_light-0","assistant","light","Turn off the light","","Sorry, I couldn't understand that","eval-test_expected_states","Bad","Sorry, I couldn't understand that,? +++++"
      "dom1_pl_lights-turn_off_the_lights_upstairs-0","assistant","light","Turn off the lights upstairs","{'tool_name': None, 'tool_args': None}","Turned off the lights","eval-test_expected_states","Bad","AssertionError: assert not {'light.bedroom_2_light': {'expected': {'color_mode': 'onoff'}, 'got': {'color_mode': None}}, 'light.bedroom_3_light':...t': {'color_mode': None}}, 'light.bedroom_4_light': {'expected': {'color_mode': 'onoff'}, 'got': {'color_mode': None}}}"
      "dom1_pl_lights-turn_off_the_living_room_light_then_turn_on_the_kitchen_light-0","assistant","light","Turn off the living room light then turn on the kitchen light","","Sorry, I am not aware of any device called off the living room light then turn on the kitchen light","eval-test_expected_states","Bad","AssertionError: assert not {'light.living_room_light': {'expected': {'state': 'off'}, 'got': {'state': 'on'}}}"
      "dom1_pl_lights-turn_on_all_the_bedroom_lights_upstairs-0","assistant","light","Turn on all the bedroom lights upstairs","","Sorry, I am not aware of any device called all the bedroom in the upstairs floor","eval-test_expected_states","Bad","AssertionError: assert not {'light.bedroom_2_light': {'expected': {'state': 'on'}, 'got': {'state': 'off'}}, 'light.bedroom_3_light': {'expected'...ate': 'on'}, 'got': {'state': 'off'}}, 'light.bedroom_4_light': {'expected': {'state': 'on'}, 'got': {'state': 'off'}}}"
      "dom1_pl_lights-turn_on_all_the_upstairs_lights-0","assistant","light","Turn on all the upstairs lights","{'tool_name': None, 'tool_args': None}","Turned on the lights","eval-test_expected_states","Bad","AssertionError: assert not {'light.bedroom_2_light': {'expected': {'color_mode': None}, 'got': {'color_mode': 'onoff'}}, 'light.bedroom_3_light':... {'color_mode': 'onoff'}}, 'light.bedroom_4_light': {'expected': {'color_mode': None}, 'got': {'color_mode': 'onoff'}}}"
      "dom1_pl_lights-turn_on_both_the_kitchen_light_and_living_room_lights-0","assistant","light","Turn on both the kitchen light and living room lights","","Sorry, I am not aware of any area called both the kitchen light and living room","eval-test_expected_states","Bad","Sorry, I am not aware of any area called both the kitchen light and living room,? +++++"
      "dom1_pl_lights-turn_on_the_kitchen_light_and_then_turn_off_the_living_room_light-0","assistant","light","Turn on the kitchen light and then turn off the living room light","","Sorry, I am not aware of any device called light and then turn off the living room light in the kitchen area","eval-test_expected_states","Bad","AssertionError: assert not {'light.living_room_light': {'expected': {'state': 'off'}, 'got': {'state': 'on'}}}"
      "dom1_pl_lights-turn_on_the_light_in_the_kitchen_then_turn_off_the_light_in_the_living_room-0","assistant","light","Turn on the light in the kitchen, then turn off the light in the living room","","Sorry, I am not aware of any floor called kitchen then turn off the light in the living room","eval-test_expected_states","Bad","AssertionError: assert not {'light.living_room_light': {'expected': {'state': 'off'}, 'got': {'state': 'on'}}}"
      "dom1_pl_lights-turn_on_the_living_room_and_kitchen_lights-0","assistant","light","Turn on the living room and kitchen lights","","Sorry, I am not aware of any floor called living room and kitchen","eval-test_expected_states","Bad","Sorry, I am not aware of any floor called living room and kitchen,? +++++"
  
    ''',
  )
# ---
# name: test_assist.2
  tuple(
    'reports-by-category.yaml',
    '''
      ---
      - category: light
        good_percent: 23.1%
        confidence_interval: 22.9%
        good: 3
        total: 13
      
  
    ''',
  )
# ---
# name: test_assist.3
  tuple(
    'reports-by-model-category.yaml',
    '''
      ---
      - model_id-category: assistant-light
        good_percent: 23.1%
        confidence_interval: 22.9%
        good: 3
        total: 13
      
  
    ''',
  )
# ---
# name: test_assist.4
  tuple(
    'reports-by-model-test-name.yaml',
    '''
      ---
      - model_id-task_name: assistant-eval-test_expect_llm_tool_call_args
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - model_id-task_name: assistant-eval-test_expect_llm_tool_call_name
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - model_id-task_name: assistant-eval-test_expect_response
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - model_id-task_name: assistant-eval-test_expected_states
        good_percent: 23.1%
        confidence_interval: 22.9%
        good: 3
        total: 13
      
  
    ''',
  )
# ---
# name: test_assist.5
  tuple(
    'reports-by-task-id.yaml',
    '''
      ---
      - task_id: dom1_pl_lights-kitchen_light_off-0
        good_percent: 100.0%
        confidence_interval: 0.0%
        good: 1
        total: 1
      - task_id: dom1_pl_lights-please_turn_off_the_light-0
        good_percent: 100.0%
        confidence_interval: 0.0%
        good: 1
        total: 1
      - task_id: dom1_pl_lights-please_turn_on_the_kitchen_light-0
        good_percent: 100.0%
        confidence_interval: 0.0%
        good: 1
        total: 1
      - task_id: dom1_pl_lights-shut_off_the_upstairs_lights_please-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_off_the_light-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_off_the_lights_upstairs-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_off_the_living_room_light_then_turn_on_the_kitchen_light-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_all_the_bedroom_lights_upstairs-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_all_the_upstairs_lights-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_both_the_kitchen_light_and_living_room_lights-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_the_kitchen_light_and_then_turn_off_the_living_room_light-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_the_light_in_the_kitchen_then_turn_off_the_light_in_the_living_room-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      - task_id: dom1_pl_lights-turn_on_the_living_room_and_kitchen_lights-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 1
      
  
    ''',
  )
# ---
# name: test_assist.6
  tuple(
    'reports-by-test-name.yaml',
    '''
      ---
      - task_name: eval-test_expect_llm_tool_call_args
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - task_name: eval-test_expect_llm_tool_call_name
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - task_name: eval-test_expect_response
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 0
      - task_name: eval-test_expected_states
        good_percent: 23.1%
        confidence_interval: 22.9%
        good: 3
        total: 13
      
  
    ''',
  )
# ---
# name: test_assist.7
  tuple(
    'reports-token-stats.yaml',
    '''
      --- []
      
  
    ''',
  )
# ---
# name: test_assist.8
  tuple(
    'reports.yaml',
    '''
      ---
      - model_id: assistant
        good_percent: 23.1%
        confidence_interval: 22.9%
        good: 3
        total: 13
      
  
    ''',
  )
# ---
